### PR TITLE
Mzuevx ac criteo block

### PR DESCRIPTION
**Pull request by mzuevx**
Work around too slow accuracy tests with DLRM models on Criteo datasets.

Key problem: prepare batch data for model inputs from record-oriented annotations consumes much more time than batch inference.
Solution: make processing batch-oriented 

Changes: 
1. Add "block" parameter to criteo_kaggle_dac converter to produce batch-oriented annotations
2. Add "block" parameter to numpy_reader to allow reading full batch input
3. Modification of DLSDKLauncher._data_to_blob() to correctly handle 3D data to 2D input layout
4. Add "block" parameter to ClassificationAdapter to allow whole batch output be transformed into single prediction object
5. Add "match" parameter to "accuracy" metric to allow sklearn.metric.accuracy_score in the context of AC "accuracy" metric 

Performance benefit from these changes is about 40x times

### CODE REVIEW

**Comment by eaidova** [comment]
jenkins ok to test

File: tools/accuracy_checker/accuracy_checker/metrics/classification.py [hunk 1]
```diff
@@ -23,9 +23,10 @@
     ArgMaxClassificationPrediction
 )
 
-from ..config import NumberField, StringField, ConfigError
+from ..config import NumberField, StringField, ConfigError, BoolField
 from .metric import PerImageEvaluationMetric
 from .average_meter import AverageMeter
+from sklearn.metrics import accuracy_score
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/classification.py:29]: please make sklearn optional or do not use it at all for such simple metric.
it is problematic dependency for some platforms

File: tools/accuracy_checker/accuracy_checker/adapters/classification.py [hunk 2]
```diff
@@ -37,13 +37,17 @@ def parameters(cls):
             'argmax_output': BoolField(
                 optional=True, default=False, description="identifier that model output is ArgMax layer"
             ),
+            'block': BoolField(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/classification.py:40]: please document all added parameters in readme

File: tools/accuracy_checker/accuracy_checker/metrics/classification.py (hunk 1 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/metrics/classification.py]: fixed

File: tools/accuracy_checker/accuracy_checker/metrics/classification.py [hunk 3]
```diff
@@ -46,27 +50,43 @@ def parameters(cls):
...
-            )
+            ),
+            'match': BoolField(optional=True, default=False)
         })
 
         return parameters
 
     def configure(self):
         self.top_k = self.get_value_from_config('top_k')
+        self.match = self.get_value_from_config('match')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/classification.py:61]: please add check that accuracy_score is not None in case when match=False

File: tools/accuracy_checker/accuracy_checker/metrics/classification.py [hunk 4]
```diff
@@ -46,27 +50,43 @@ def parameters(cls):
...
+        else:
+            self.accuracy = []
 
     def update(self, annotation, prediction):
-        return self.accuracy.update(annotation.label, prediction.top_k(self.top_k))
+        if not self.match:
+            return self.accuracy.update(annotation.label, prediction.top_k(self.top_k))
+        else:
+            accuracy = accuracy_score(annotation.label, prediction.label)
+            self.accuracy.append(accuracy)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/classification.py:76]: we need to return metric for single sample in quantization case
```suggestion
            self.accuracy.append(accuracy)
            return accuracy
```

File: tools/accuracy_checker/accuracy_checker/data_readers/data_reader.py [hunk 5]
```diff
@@ -396,11 +401,14 @@ def read(self, data_id):
...
-                recno = int(recno)
-                start, _ = Path(data_id).name.split('.')
-                start = int(start)
-                return data[key[0]][recno - start, :]
+                if self.block:
+                    return data[key[0]]
+                else:
+                    recno = field_id.split('_')[-1]
+                    recno = int(recno)
+                    start, _ = Path(data_id).name.split('.')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_readers/data_reader.py:409]: it is not safe for case when file have several suffixes
```suggestion
                    start = Path(data_id).name.split('.')[0]
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py [hunk 6]
```diff
@@ -104,14 +107,24 @@ def convert(self, check_content=False, **kwargs):
...
+                        "lS_o_{}{}{}".format(i, self.separator, c_file),
                     ],
-                    y[j, ...]
+                    y[i:i+self.batch, ...]
                 ))
+            else:
+                for j in range(i, i + self.batch):
+                    annotations.append(ClassificationAnnotation(
+                        [
+                            "input.1_{}{}{}".format(j, self.separator, c_file),
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py:123]: I right understand it is model specific input name? Or is it common?

Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py:123]: Yes, model specific name. Are you want to map this in converter parameters?

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py:123]: on my view better keep it neutral in the identifier and map in config using value for specific input layer.
Who knows, maybe in future we will get new model with different input name

Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py:123]: Ok

Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py]: Fixed

File: tools/accuracy_checker/accuracy_checker/metrics/classification.py (hunk 3 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/metrics/classification.py:66]: fixed

File: tools/accuracy_checker/accuracy_checker/adapters/classification.py (hunk 2 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/adapters/classification.py:40]: Fixed

**Review by eaidova — APPROVED** [review]
